src/main/java/org/opensearch/knn/jni/FaissService.java

```diff
@@ -181,6 +181,21 @@ public static native KNNQueryResult[] queryIndexWithFilter(
      */
     public static native long transferVectors(long vectorsPointer, float[][] trainingData);
 
+    /**
+     * Transfer vectors from Java to native layer. This is the version 2 of transfer vector functionality. The
+     * difference between this and the version 1 is, this version puts vectors at the end rather than in front.
+     * Keeping this name as V2 for now, will come up with better name going forward.
+     * <p>
+     * TODO: Rename the function
+     * <br>
+     * TODO: Make this function native function and use a common cpp file to host these functions.
```

Comment:
What does this mean?

Reply:
I added some todos on this function as this is not ready for production use. 

One todo is to rename and another is, as this function will be used for both nmslib and faiss. Having a function transfervectorv2 in faiss service doesn’t make sense. I want to create a common cpp class and common java class to access that function. Hence added this todo. Will resolve this in upcoming prs